File: apps/AI-Keyboard/Android/app/src/main/java/ai/zetic/demo/cherrypad/model/Stance.kt
package ai.zetic.demo.cherrypad.model

/** Reply stances. `descriptor` is injected into the reply prompt. */
enum class Stance(val label: String, val descriptor: String) {
    AGREEABLE("Agreeable", "agreeable and supportive"),
    DISAGREEABLE("Disagreeable", "politely disagreeing"),
}

File: apps/AI-Keyboard/iOS/Shared/Stance.swift
import Foundation

/// Reply stance offered as chips under the Reply action.
enum Stance: String, Codable, CaseIterable, Identifiable, Hashable {
    case agreeable
    case disagreeable

    var id: String { rawValue }

    /// Chip label.
    var label: String {
        switch self {
        case .agreeable:    return "Agreeable"
        case .disagreeable: return "Disagreeable"
        }
    }

    /// Phrase injected into the reply prompt.
    var descriptor: String {
        switch self {
        case .agreeable:    return "agreeable and supportive"
        case .disagreeable: return "politely disagreeing"
        }
    }
}
